src/components/molecules/SfInputNumber/SfInputNumber.js

```diff
@@ -0,0 +1,201 @@
+            return str.replace(/\B(?=(\d{3})+(?!\d))/g, this.thousands);
+          } else {
+            return str;
+          }
+        }
+        return this.currentValue;
+      },
+      set(value) {
+        if ((value !== null && value !== '') || this.autoFill) {
+          let num = this.parseNumber(value);
+          if (isNaN(num)) {
+            num = 0;
+          }
+          // force current number value to sync with input value
+          this.currentValue = num;
+          // check if number is valid
+          this.setCurrentValue(num);
+        } else {
+          // sey empty
+          this.setCurrentValue(null);
+        }
+      }
+    }
+  },
+
+  methods: {
+    parseNumber(value) {
+      switch (typeof value) {
+        case "string":
+          if (typeof this.thousands === "string" && this.thousands !== "") {
+            // remove thousands separator first
+            value = value.replace(new RegExp(this.thousands, "g"), "");
+          }
+          if (this.precision) {
+            let parts = value.split(this.delimiter, 2);
+            if (parts.length === 1) {
+              // no decimals
+              return parseInt(value);
+            }
+            // fix up to maximum precision on decimal places
+            let decimalPlaces = parts[1].length;
+            if (decimalPlaces > this.precision) {
+              // user may have added digit at the end of input
+              let digits = parts[0] + parts[1];
+              value =
+                digits.slice(0, -this.precision) +
+                "." +
+                digits.slice(-this.precision);
+            } else if (this.delimiter !== ".") {
+              // just fix the delimiter before parsing to float
+              value = value.replace(this.delimiter, ".");
+            }
+            return parseFloat(value);
+          }
+          // no decimal precision: integer value
+          return parseInt(value, 10);
+        case "number":
+          if (!isNaN(value)) {
+            return value;
+          }
+      }
+      return this.min || 0;
+    },
+
+    keypress(e) {
+      if (e && e.key) {
+        switch (e.key) {
+          case "0":
+          case "1":
+          case "2":
+          case "3":
+          case "4":
+          case "5":
+          case "6":
+          case "7":
+          case "8":
+          case "9":
+          case "Enter":
+            return true;
+          case this.delimiter:
+            if (this.precision) {
+              return true;
+            }
+        }
+        e.preventDefault();
+      }
+    },
+
+    setCurrentValue(val) {
+      if (typeof val === 'number') {
+        // check minimum and maximum
+        if (typeof this.min === "number" && val < this.min) {
+          val = this.min;
+        } else if (typeof this.max === "number" && val > this.max) {
+          val = this.max;
+        }
+      }
+      this.currentValue = val;
+      this.$emit("change", val);
```

Comment:
@filrak my fault here...
must be `this.$emit("input", val);` to handle `v-model` properly, can you change it directly (or may I change)? or should I create another pr?